julia-mode.el

```diff
@@ -919,6 +941,42 @@ buffer where the LaTeX symbol starts."
           (abbrev-insert symb name beg end)))
     #'ignore))
 
+(defun julia-mode--latexsub-before-point ()
+  "When there is a LaTeX substitution that can be made before the point, return (CONS BEG LATEX).
+
+`beg' is the position of the `\`, `latex' is the string to replace, including the `\`.
+
+When multiple options match, ask the user to clarify via `julia-latexsub-selector', unless there is a complete match and `julia-latexsub-greedy' is `t'."
+  (when-let (beg (julia--latexsub-start-symbol))
+    (let ((partial (buffer-substring-no-properties beg (point))))
+      (when-let (replacements (gethash partial julia-mode--latexsubs-partials))
+        (let* ((complete-match (member partial replacements))
+               (latex (cond
+                       ;; complete match w/ greedy
+                       ((and complete-match julia-latexsub-greedy) partial)
+                       ;; multiple replacements, ask user
+                       ((cdr replacements) (funcall julia-latexsub-selector replacements))
+                       ;; single replacement, pick that
+                       (t (car replacements)))))
+          (cons beg latex))))))
+
+(defun julia-latexsub-or-indent (arg)
+  "Either indent according to Julia mode conventions or perform a LaTeX-like symbol substution.
+
+When multiple options match, ask the user to clarify via `julia-latexsub-selector', unless there is a complete match and `julia-latexsub-greedy' is `t'.
+
+Presently, this is not the default. Enable with eg
+
+(define-key julia-mode-map (kbd \"TAB\") 'julia-latexsub-or-indent)
+
+eg in your `julia-mode-hook'."
+  (interactive "*i")
+  (if-let (replacement (julia-mode--latexsub-before-point))
+      (progn
+        (delete-backward-char (- (point) (car replacement)))
+        (insert (gethash (cdr replacement) julia-mode-latexsubs)))
+    (julia-indent-line)))
```

Comment:
Wait, if this ultimately calls `julia-indent-line`, I don't think it solves the problem from #193 at all. The issue in #193 was being able to have `TAB` work for both latex substitution as well as inserting `TAB`s (see example in this comment). I think this might do what ronisbr was looking for if it called `indent-for-tab-command` instead of `julia-indent-line`.

Reply:
Sorry, I missed this comment, just found it among messages. This PR solves the problem for me, if anyone still has problems with it please open another issue.

Specifically, it only calls `julia-indent-line` when it could not make a substitution. So, TAB tries to subtitute, and indents if it can't.  I find that behavior useful.

Reply:
So basically it gives you the reverse of the behavior of `indent-for-tab-command` when `tab-always-indent` is set to `'complete`:

- With `indent-for-tab-command` with `(setq tab-always-indent 'complete)`: indent then complete
- With this function: complete (but only latexsubs) then indent

If we change `julia-indent-line` in this function to `indent-for-tab-command`, the user can change between TAB inserting tabs and TAB indenting the line by customizing the value of `tab-always-indent`. I'll create a PR for your review when I have a few minutes.

If you haven't seen it yet, could you please comment on [my other review comment when you get a chance](https://github.com/JuliaEditorSupport/julia-emacs/pull/196#discussion_r1592636480)?